dsl/camel-jbang/CamelJbang.java

```diff
@@ -0,0 +1,252 @@
+        CamelContext context = main.getCamelContext();
+
+        try (GitHubResourceResolver resolver = new GitHubResourceResolver()) {
+            resolver.setCamelContext(context);
+
+            Resource resource = resolver.resolve("github:apache:camel-kamelets:docs/modules/ROOT/nav.adoc");
+
+            if (!resource.exists()) {
+                throw new Exception("The resource does not exist");
+            }
+
+            try (FileWriter w = new FileWriter(indexFile)) {
+                Reader resourceReader = resource.getReader();
+
+                resourceReader.transferTo(w);
+            }
+        }
+    }
+
+    /*
+     * Matches the following line. Separate them into groups and pick the last
+     * which contains the description:
+     *
+     * xref:ROOT:mariadb-sink.adoc[image:kamelets/mariadb-sink.svg[] MariaDB Sink]
+     */
+    private void checkForMatches(String line) {
+        Pattern pat = Pattern.compile("(.*):(.*):(.*)\\[(.*)\\[\\] (.*)\\]");
+
+        Matcher matcher = pat.matcher(line);
+
+        if (matcher.find()) {
+            String description = matcher.group(5);
+            String kamelet = matcher.group(3).replace(".adoc", "");
+
+            if (description.toLowerCase().contains(searchTerm.toLowerCase())) {
+                System.out.printf("%-35s %-45s%n", kamelet, description);
+            }
+        }
+    }
+
+    private void printAll(String line) {
+        Pattern pat = Pattern.compile("(.*):(.*):(.*)\\[(.*)\\[\\] (.*)\\]");
+
+        Matcher matcher = pat.matcher(line);
+
+        if (matcher.find()) {
+            String kamelet = matcher.group(3).replace(".adoc", "");
+            String description = matcher.group(5);
+
+            System.out.printf("%-35s %-45s%n", kamelet, description);
+        }
+    }
+
+    private void readFileByLine(File indexFile, Consumer<String> consumer) throws IOException, FileNotFoundException {
+        FileReader indexFileReader = new FileReader(indexFile);
+        try (BufferedReader br = new BufferedReader(indexFileReader)) {
+
+            String line;
+            do {
+                line = br.readLine();
+
+                if (line == null) {
+                    break;
+                }
+
+                consumer.accept(line);
+
+            } while (line != null);
+        }
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        File indexFile = new File("index");
+        indexFile.deleteOnExit();
+
+        downloadResource(indexFile);
+
+        System.out.printf("%-35s %-45s%n", "KAMELET", "DESCRIPTION");
+        System.out.printf("%-35s %-45s%n", "-------", "-----------");
+        if (!searchTerm.isEmpty()) {
+            readFileByLine(indexFile, this::checkForMatches);
+        } else {
+            readFileByLine(indexFile, this::printAll);
+        }
+
+        return 0;
+    }
+}
+
+@Command(name = "CamelJbang", mixinStandardHelpOptions = true, version = "CamelJbang 3.12.0-SNAPSHOT",
```

Comment:
fyi, its JBang, jbang is the command, Jbang is a mistake :)

Reply:
Thanks. I think I corrected it now :)